tests/protocols/test_websocket.py

```diff
@@ -466,3 +466,32 @@ async def get_subprotocol(url):
         accepted_subprotocol = loop.run_until_complete(get_subprotocol(url))
         assert accepted_subprotocol == subprotocol
         loop.close()
+
+
+@pytest.mark.parametrize("protocol_cls", WS_PROTOCOLS)
+def test_server_lost_connection(protocol_cls, reraise):
+    class App(WebSocketResponse):
+        async def websocket_connect(self, message):
+            await self.send({"type": "websocket.accept"})
+            await self.receive()
+
+            # Simulate a lost connection
+            # without receiving a close frame
+            self.send.__self__.connection_lost(None)
```

Comment:
I'm still wondering if there's a better way to access the underlying protocol here.

Reply:
yes, you're right.
add to that the fact that once we call the protocol here, right after in the `asgi_send` mehtod at `elif not self.closed_event.is_set():` we have that event not set and the State that is `State.CLOSED` which is dubious

Reply:
fail_connection ?

Reply:
I used `connection_lost` because it was used in [`test_early_disconnect`](https://github.com/encode/uvicorn/blob/ae0fd316f03dbef926e40216024dfb934417d48d/tests/protocols/test_http.py#L487) in the http tests. It's the method called on protocol from [transport.close()](https://docs.python.org/3/library/asyncio-protocol.html#asyncio.BaseTransport.close), so seems appropriate. I was just wondering if there's a nicer way to access it instead of going through the methods `__self__.